modules/tinymce/src/core/main/ts/DragDropOverrides.ts

```diff
@@ -33,8 +38,16 @@ interface State {
   width: number;
   height: number;
   ghost: HTMLElement;
+  intervalId: Optional<number>;
 }
 
+const clearStateInterval = (state: State) => {
+  if (state.intervalId.isSome()) {
+    clearInterval(state.intervalId.getOrUndefined());
+    state.intervalId = Optional.none();
+  }
+};
```

Comment:
We have a concept called `Singleton` that seems like it might make this easier - it ensures a "revoke" function is run whenever a new value is set, which could be used to clear the interval. No need to manually manage set and clear.
https://github.com/tinymce/tinymce/blob/2d8b95517a43d1f1967a2f3898371865a2fa18e4/modules/katamari/src/main/ts/ephox/katamari/api/Singleton.ts#L22

I'm not sure it _quite_ applies here, because currently it is designed for objects with `destroy` or `unbind` functions, but I think the concept might be useful. It would certainly cut down on some of the boilerplate code you've had to add here.

Perhaps we need a new singleton that supports intervals like this; it might be a good candidate to go in the [Throttler](https://github.com/tinymce/tinymce/blob/2d8b95517a43d1f1967a2f3898371865a2fa18e4/modules/katamari/src/main/ts/ephox/katamari/api/Throttler.ts) module. Throttler uses timeout, not interval, but it seems like the most appropriate place.

Reply:
Done!